From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 28 Jan 2016 20:24:28 +0100
Subject: nspawn: make sure --link-journal=host may be used twice in a row

Fixes #2186

This fixes fall-out from 574edc90066c3faeadcf4666928ed9b0ac409c75.

(cherry-picked from commit ba8e6c4d0e074a3df0b55751f1753460ce82f1eb)
---
 src/nspawn/nspawn.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/nspawn/nspawn.c b/src/nspawn/nspawn.c
index f6a2c03..d28f41b 100644
--- a/src/nspawn/nspawn.c
+++ b/src/nspawn/nspawn.c
@@ -1457,8 +1457,7 @@ static int setup_journal(const char *directory) {
         if (arg_link_journal == LINK_HOST) {
                 /* don't create parents here -- if the host doesn't have
                  * permanent journal set up, don't force it here */
-                r = mkdir(p, 0755);
-                if (r < 0) {
+                if (mkdir(p, 0755) < 0 && errno != EEXIST) {
                         if (arg_link_journal_try) {
                                 log_debug_errno(errno, "Failed to create %s, skipping journal setup: %m", p);
                                 return 0;
